tests/e2e.sh

```diff
@@ -127,9 +166,14 @@ echo "access kube-state-metrics metrics endpoint"
 curl -s "http://localhost:8001/api/v1/proxy/namespaces/kube-system/services/kube-state-metrics:8080/metrics" >$KUBE_STATE_METRICS_LOG_DIR/metrics
 
 echo "check metrics format with promtool"
-wget -q -O /tmp/prometheus.tar.gz https://github.com/prometheus/prometheus/releases/download/v$PROMETHEUS_VERSION/prometheus-$PROMETHEUS_VERSION.linux-amd64.tar.gz
-tar zxfv /tmp/prometheus.tar.gz -C /tmp
-cat $KUBE_STATE_METRICS_LOG_DIR/metrics | /tmp/prometheus-$PROMETHEUS_VERSION.linux-amd64/promtool check metrics
+[ -n "$E2E_SETUP_PROMTOOL" ] && setup_promtool
+cat $KUBE_STATE_METRICS_LOG_DIR/metrics | promtool check metrics
+
+collectors="configmap cronjob daemonset deployment endpoint hpa job limitrange namespace node persistentvolumeclaim pod replicaset replicationcontroller resourcequota secret service statefulset"
```

Comment:
Maybe we should generate the collectors list from collectors package by leveraging the go source file names. This help us from avoids updating e2e test when new resource is added or deleted.

Reply:
Ok what about
```
$ find collectors/ -name "*.go" -not -name "*_test.go" -not -name "collectors.go" | xargs -n1 basename | awk -F. '{print $1}'
```

Reply:
seems good to me.

Reply:
Addessed in 0e48a35.